config_moodle.txt
Instalação do MoodleMat:
1) pacotes: apt-get install graphviz aspell php5-pspell php5-curl php5-gd php5-intl php5-mysql php5-xmlrpc php5-ldap

Configurações do MoodleMat:
1) Home -> My profile settings -> Edit profile
a) Email address: moodlemat@mat.unb.br

2) Mudar senha do admin: My profile settings -> Change password

2) Site administration -> Plugins -> Message outputs -> Email
a) SMTP hosts: mx1.mat.unb.br
b) SMTP username:
c) No-reply address: noreply@mat.unb.br

2) Site administration -> Server -> Support contact
a) Support email: moodlemat@mat.unb.br

3) Permissões de novos Usuários: Site administration -> Users -> Permissions -> Define roles
a) Na aba 'Allow role assignments' ir na linha "Teacher" marcar a coluna "Teacher"

4) Configurações da página inicial e título do Moodle: Site administration -> Front page settings -> Edit settings
a) Full site name: Moodle MAT
b) Short name for site: MoodleMAT
c) Front page items when logged in: Enrolled courses

5) Página de login: Site administration -> Plugins -> Authentication -> Manage authentication
a) Rolar a página até o campo "Instructions" e acrescentar o texto:
"
Caro usuário, se você já é aluno de alguma disciplina do MAT e ainda não efetuou login no MoodleMat, clique aqui(link em vermelho para http://e-mural.mat.unb.br/cadastro.php)!

Nesta página preencha os dados solicitados e informe um endereço de e-mail. Após salvar seu cadastro você receberá uma mensagem de confirmação em seu e-mail."
"

6) Instalação do pt_br:
a) Site administration -> Language -> Language packs: Instalar "Português Brasil"
b) Site administration -> Language -> Language settings: Default language: "Português Brasil"

7) Criação do papel de Monitor: Site administration -> Users -> Permissions -> Define roles
a) Short name: Monitor
b) Custom full name: Estudante(Monitor)
c) Custom description: Perfil para incluir monitores nos cursos.
d) Em 'Filter' pesquisar por 'grade' e desabilitar as seguintes opções:
a1) Activity: Assignment
Grade assignment
mod/assign:grade
 Allow
XSS risk
Review and release grades
mod/assign:managegrades
 Allow
Release grades
mod/assign:releasegrades
 Allow
Review grades
mod/assign:reviewgrades
 Allow
Activity: Assignment (2.2)
Grade assignment
mod/assignment:grade
 Allow
XSS risk
Activity: LTI
View grades returned by the external tool
mod/lti:grade
 Allow
Privacy risk
Activity: Quiz
Grade quizzes manually
mod/quiz:grade
 Allow
Spam risk
Regrade quiz attempts
mod/quiz:regrade
 Allow
Spam risk
Activity: Workshop
Override calculated grades
mod/workshop:overridegrades


8) Site administration -> Plugins -> Filters -> Manage filters
a) Ativar a opção TeX Notation
b) Ir em Settings e mudar os caminhos:
- LaTeX preamble: \usepackage[utf8]{inputenc}
- Path of latex binary: /usr/bin/latex
- Path of dvips binary: /usr/bin/dvips
- Path of convert binary: /usr/bin/convert
- convert output format: PNG

9) Site administration -> Courses -> Backups -> General backup defaults
a) Desmarcar opção "Include users"


10) Site administration -> Users -> Permissions -> Define roles: Na aba 'Manage roles'
a) Editar o papel de professor e desmarcar 'Manual enrolments -> Unenrol users from the course'
b) Editar o papel de professor e desmarcar 'Unenrol users from the course enrol -> manual:unenrol'

11) Abrir link da senha esquecida em nova janela: Editar arquivo moodle/login/index_form.html e trocar a linha

<div class="forgetpass"><a href="forgot_password.php"><?php print_string("forgotten") ?></a></div>

Por

<div class="forgetpass"><a target = "_blank" href="<?php echo $CFG->wwwemural ?>esqueci_senha?moodle"><?php print_string("forgotten") ?></a></div>


12) Para permitir a solicitação da criação de novos cursos: Site administration -> Courses -> Course request
a) Marcar a opção 'Enable course requests'
b) Mudar para usuário 'Admin' em 'Course request notification'

13) Para aprovar ou não a criação de um curso: Site administration -> Courses -> Manage courses and categories
a) Clicar no link 'Courses pending approval'

14) Criação de role que permitir criar usuários manualmente: Site administration -> Users -> Permissions -> Define roles:
a0) Short name: co-admin
a00) Custom full name: Co-administrador
a) Em 'Manage roles' ir em 'Add a new role'
b) Em 'Use role or archetype' escolher 'Manager'
c) Em 'Context types where this role may be assigned' marcar 'System, User, Category, Course'
d) Em 'Allow role assignments' selecionar Teacher, Estudante(Monitor) e Student
e) Em 'Allow role overrides' selecionar 'Teacher, Estudante(Monitor) e Student'
f) Em 'Allow role switches' selecionar 'Teacher, Estudante(Monitor) e Student'
g) Nas permissões do papel desmarcar tudo com excessão de: View participants, Create users, Configure manual enrol instances, Enrol users, Manage user enrolments, Unenrol users from the course, Manage enrolled users, View hidden courses, View participants.
h) Marcar opções: Update user profiles, Edit user profile

15) Criar usuário 'coadmin' com nome "Co-administrador Ajudante" com e-mail falso: 'coadmin@mat.unb.br'.

16) Depois ir em Site administration -> Users -> Permissions -> Assign system roles e colocar na role 'Co-Admin' o usuário 'coadmin'.

17) Configuração abertura de abas do do bloco de navegação de cursos: ir em Home -> Navigation -> Configuration
a) Em 'Generate navigation for the following' escolher 'Categories and courses'

Configurações gerais
1) Criar pasta "anosemestre" no formato "yyyyz", onde Z = 1 ou 2, na pasta "/moodledata"
a) Usar pasta 'moodledata-BASE' para criar as demais.
b) Permissões 777, dono e grupo www-data.

2) Para reativar o tema logue como "admin" e vá em: "Site administration -> Appearance -> Theme selector" e clicar em "Clear theme caches".


3) Para alterar comportamento do 'restore' de cursos:
a) Acessar arquivo 'moodle/backup/util/ui/renderer.php' e mudar as linhas:
$html .= $this->backup_detail_input(get_string('restoretocurrentcourseadding', 'backup'), 'radio', 'target', backup::TARGET_CURRENT_ADDING, array('checked'=>'checked'));
$html .= $this->backup_detail_input(get_string('restoretocurrentcoursedeleting', 'backup'), 'radio', 'target',backup::TARGET_CURRENT_DELETING);

por

$html .= $this->backup_detail_input(get_string('restoretocurrentcourseadding', 'backup'), 'radio', 'target', backup::TARGET_CURRENT_ADDING);
            $html .= $this->backup_detail_input(get_string('restoretocurrentcoursedeleting', 'backup'), 'radio', 'target', backup::TARGET_CURRENT_DELETING, array('checked'=>'checked'));

O efeito da mudança acima é colocar como default o restore deletando o curso atual.


b) Acessar arquivo 'moodle/backup/moodle2/restore_course_task.class.php' e mudar a linha:

$keep_enrols = new restore_course_generic_setting('keep_roles_and_enrolments', base_setting::IS_BOOLEAN, false);

para

$keep_enrols = new restore_course_generic_setting('keep_roles_and_enrolments', base_setting::IS_BOOLEAN, true);

e a linha

$keep_groups = new restore_course_generic_setting('keep_groups_and_groupings', base_setting::IS_BOOLEAN, false);

para

$keep_groups = new restore_course_generic_setting('keep_groups_and_groupings', base_setting::IS_BOOLEAN, true);

O efeito das mudanças acima é manter os usuários e grupos que já tiverem sido criados pelo E-mural.

c) Para ocultar as opções sobreescrever usuários e grupos: abrir arquivo 'moodle/backup/moodle2/restore_course_task.class.php'e acrescentar:

c1) Antes da linha

if ($this->get_target() != backup::TARGET_CURRENT_DELETING and $this->get_target() != backup::TARGET_EXISTING_DELETING) {

O código abaixo:

$keep_enrols->set_visibility(backup_setting::HIDDEN);

c2) Antes da linha

if ($this->get_target() != backup::TARGET_CURRENT_DELETING and $this->get_target() != backup::TARGET_EXISTING_DELETING) {

O código abaixo:

$keep_groups->set_visibility(backup_setting::HIDDEN);

c3) Antes da linha:

if ($this->get_target() == backup::TARGET_NEW_COURSE) {

O código abaixo:

$overwrite->set_visibility(backup_setting::HIDDEN);

4)

Acrescetar no crontab a linha
*/30 * * * * /usr/bin/php  /var/www/moodle/admin/cli/cron.php >/dev/null

para envio de e-mail.

5) Na versão 3.0 do Moodle, deixar vazio o campo 'SMTP username' localizado em "Site administration ->︎ Plugins ->︎ Message outputs ->︎ Email".

6) Copiar pasta 'emural' em 'plugins' para '/var/www/moodle/auth/'

7) Copiar pasta 'staticpage' em 'plugins' para '/var/www/moodle/local/'

8) Copiar pasta 'staticpage' em 'moodledata-BASE' para '/moodledata/anosemestre/'

9) Na configuração do MathJax incluir:

MathJax.Hub.Config({
    config: ["Accessible.js", "Safe.js"],
    tex2jax: {
      inlineMath: [['$','$'], ['\\(','\\)']],
      displayMath: [ ['$$','$$'],["\\[","\\]"] ],
      processEscapes: true,
      TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
    },
    errorSettings: { message: ["!"] },
    skipStartupTypeset: true,
    messageStyle: "none"
});

10) Em "Site administration ->︎ Plugins ->︎ Filters ->︎ Manage" destivar o 'Tex Notation'

11) Para atualizar o endereço dos arquivos quando mudar o link usar o endereço http://moodle.mat.unb.br/admin/tool/replace/index.php.

12) Acessar como 'admin' ir em 'Administration' e clicar em 'Turn editing on'. Após isso editar as configurações de navegação e no campo 'Generate navigation for the following' escolhar 'Categories and courses'.

13) Ir em 'Frontpage settings' e trocar 'Front page' para 'List of categories'.

14) Ao atualizar o tema KLASS ir em 'theme/klass/scss/preset/theme.scss' e trocar o valor da variável $color_thunderbird_approx para $color_thunderbird_approx: #00923f;

15) Para o STACK, reinstalar seguinte os passos aqui: https://docs.stack-assessment.org/en/Installation/

16) Após, seguir os passos em https://docs.stack-assessment.org/en/CAS/Optimising_Maxima/, na sessão "GCL", lembrando de primeiro executar o script de checagem usando como padrão 'Linux' e com o campo "Optimised Maxima command" em branco.

17) Após a instalação do Stack verificar a página https://docs.stack-assessment.org/en/Installation/Testing_installation/ para testar a instalação.

16) O path para ser usado no maxima é sempre "/moodledata/anosemestre/stack".

19) Após isso mudar, nas configurações do STACK, a plataforma para "Linux (optimised)", Maxima version para "default" e Optimised Maxima command para "/moodledata/anosemestre/stack/maxima-optimised  -eval '(cl-user::run)'"

20) Copiar pasta 'reminders' em 'local' para '/var/www/moodle/local/'

21) Copiar pasta 'userenrols' em 'local' para '/var/www/moodle/local/'

22) Copiar pasta 'klass' em 'theme' para '/var/www/moodle/theme/'

23) Copiar pasta 'attendance' em 'mod' para '/var/www/moodle/mod/'

24) Copiar pasta 'hvp' em 'mod' para '/var/www/moodle/mod/'

25) Copiar pasta 'ipaddress' em 'mod/quizz/accessrule' para '/var/www/moodle/mod/quizz/accessrule'

26) Copiar pasta 'ipaddresslist' em 'mod/quizz/accessrule' para '/var/www/moodle/mod/quizz/accessrule'

27) Copiar pasta 'onesession' em 'mod/quizz/accessrule' para '/var/www/moodle/mod/quizz/accessrule'

28) Copiar pasta 'ddmatch' em 'question/type' para '/var/www/moodle/question/type'

29) Copiar pasta 'gapfill' em 'question/type' para '/var/www/moodle/question/type'

30) Copiar pasta 'shortmath' em 'question/type' para '/var/www/moodle/question/type'

31) Copiar pasta 'stack' em 'question/type' para '/var/www/moodle/question/type'

32) Copiar pasta 'varnumeric' em 'question/type' para '/var/www/moodle/question/type'

33) Copiar pasta 'varnumericset' em 'question/type' para '/var/www/moodle/question/type'

34) Copiar pasta 'adaptivemultipart' em 'question/behaviour' em 'question/type' para '/var/www/moodle/question/behaviour'

35) Copiar pasta 'dfcbmexplicitvaildate' em 'question/behaviour' para '/var/www/moodle/question/behaviour'

36) Copiar pasta 'dfexplicitvaildate' em 'question/behaviour' para '/var/www/moodle/question/behaviour'

37) Copiar pasta 'examview' em 'question/format' para '/var/www/moodle/question/format'


MoodleMat/MOODLE-BASE/muc/config.php
<?php defined('MOODLE_INTERNAL') || die();
 $configuration = array (
  'siteidentifier' => '0ca7dbfc323e9ba318dface4b2878822',
  'stores' => 
  array (
    'default_application' => 
    array (
      'name' => 'default_application',
      'plugin' => 'file',
      'configuration' => 
      array (
      ),
      'features' => 14,
      'modes' => 3,
      'default' => true,
      'class' => 'cachestore_file',
      'lock' => 'cachelock_file_default',
    ),
    'default_session' => 
    array (
      'name' => 'default_session',
      'plugin' => 'session',
      'configuration' => 
      array (
      ),
      'features' => 14,
      'modes' => 2,
      'default' => true,
      'class' => 'cachestore_session',
      'lock' => 'cachelock_file_default',
    ),
    'default_request' => 
    array (
      'name' => 'default_request',
      'plugin' => 'static',
      'configuration' => 
      array (
      ),
      'features' => 14,
      'modes' => 4,
      'default' => true,
      'class' => 'cachestore_static',
      'lock' => 'cachelock_file_default',
    ),
  ),
  'modemappings' => 
  array (
    0 => 
    array (
      'mode' => 1,
      'store' => 'default_application',
      'sort' => -1,
    ),
    1 => 
    array (
      'mode' => 2,
      'store' => 'default_session',
      'sort' => -1,
    ),
    2 => 
    array (
      'mode' => 4,
      'store' => 'default_request',
      'sort' => -1,
    ),
  ),
  'definitions' => 
  array (
    'core/string' => 
    array (
      'mode' => 1,
      'simplekeys' => true,
      'simpledata' => true,
      'staticacceleration' => true,
      'staticaccelerationsize' => 30,
      'component' => 'core',
      'area' => 'string',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/langmenu' => 
    array (
      'mode' => 1,
      'simplekeys' => true,
      'simpledata' => true,
      'staticacceleration' => true,
      'component' => 'core',
      'area' => 'langmenu',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/databasemeta' => 
    array (
      'mode' => 1,
      'requireidentifiers' => 
      array (
        0 => 'dbfamily',
      ),
      'staticacceleration' => true,
      'staticaccelerationsize' => 15,
      'component' => 'core',
      'area' => 'databasemeta',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/eventinvalidation' => 
    array (
      'mode' => 1,
      'staticacceleration' => true,
      'requiredataguarantee' => true,
      'simpledata' => true,
      'component' => 'core',
      'area' => 'eventinvalidation',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/questiondata' => 
    array (
      'mode' => 1,
      'simplekeys' => true,
      'requiredataguarantee' => false,
      'datasource' => 'question_finder',
      'datasourcefile' => 'question/engine/bank.php',
      'component' => 'core',
      'area' => 'questiondata',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/htmlpurifier' => 
    array (
      'mode' => 1,
      'component' => 'core',
      'area' => 'htmlpurifier',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/config' => 
    array (
      'mode' => 1,
      'staticacceleration' => true,
      'simpledata' => true,
      'component' => 'core',
      'area' => 'config',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/groupdata' => 
    array (
      'mode' => 1,
      'simplekeys' => true,
      'simpledata' => true,
      'staticacceleration' => true,
      'staticaccelerationsize' => 2,
      'component' => 'core',
      'area' => 'groupdata',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/calendar_subscriptions' => 
    array (
      'mode' => 1,
      'simplekeys' => true,
      'simpledata' => true,
      'staticacceleration' => true,
      'component' => 'core',
      'area' => 'calendar_subscriptions',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/yuimodules' => 
    array (
      'mode' => 1,
      'component' => 'core',
      'area' => 'yuimodules',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/observers' => 
    array (
      'mode' => 1,
      'simplekeys' => true,
      'simpledata' => true,
      'staticacceleration' => true,
      'staticaccelerationsize' => 2,
      'component' => 'core',
      'area' => 'observers',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/plugin_manager' => 
    array (
      'mode' => 1,
      'simplekeys' => true,
      'simpledata' => true,
      'component' => 'core',
      'area' => 'plugin_manager',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/coursecattree' => 
    array (
      'mode' => 1,
      'staticacceleration' => true,
      'invalidationevents' => 
      array (
        0 => 'changesincoursecat',
      ),
      'component' => 'core',
      'area' => 'coursecattree',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/coursecat' => 
    array (
      'mode' => 2,
      'invalidationevents' => 
      array (
        0 => 'changesincoursecat',
        1 => 'changesincourse',
      ),
      'ttl' => 600,
      'component' => 'core',
      'area' => 'coursecat',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/coursecatrecords' => 
    array (
      'mode' => 4,
      'simplekeys' => true,
      'invalidationevents' => 
      array (
        0 => 'changesincoursecat',
      ),
      'component' => 'core',
      'area' => 'coursecatrecords',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/coursecontacts' => 
    array (
      'mode' => 1,
      'staticacceleration' => true,
      'simplekeys' => true,
      'component' => 'core',
      'area' => 'coursecontacts',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/repositories' => 
    array (
      'mode' => 4,
      'component' => 'core',
      'area' => 'repositories',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/externalbadges' => 
    array (
      'mode' => 1,
      'simplekeys' => true,
      'ttl' => 3600,
      'component' => 'core',
      'area' => 'externalbadges',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/coursemodinfo' => 
    array (
      'mode' => 1,
      'simplekeys' => true,
      'component' => 'core',
      'area' => 'coursemodinfo',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/userselections' => 
    array (
      'mode' => 2,
      'simplekeys' => true,
      'simpledata' => true,
      'component' => 'core',
      'area' => 'userselections',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/gradecondition' => 
    array (
      'mode' => 1,
      'staticacceleration' => true,
      'staticaccelerationsize' => 2,
      'ttl' => 3600,
      'component' => 'core',
      'area' => 'gradecondition',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/navigation_expandcourse' => 
    array (
      'mode' => 2,
      'simplekeys' => true,
      'simpledata' => true,
      'component' => 'core',
      'area' => 'navigation_expandcourse',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'core/suspended_userids' => 
    array (
      'mode' => 4,
      'simplekeys' => true,
      'simpledata' => true,
      'component' => 'core',
      'area' => 'suspended_userids',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'repository_skydrive/foldername' => 
    array (
      'mode' => 2,
      'component' => 'repository_skydrive',
      'area' => 'foldername',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
    'tool_uploadcourse/helper' => 
    array (
      'mode' => 4,
      'component' => 'tool_uploadcourse',
      'area' => 'helper',
      'selectedsharingoption' => 2,
      'userinputsharingkey' => '',
    ),
  ),
  'definitionmappings' => 
  array (
  ),
  'locks' => 
  array (
    'cachelock_file_default' => 
    array (
      'name' => 'cachelock_file_default',
      'type' => 'cachelock_file',
      'dir' => 'filelocks',
      'default' => true,
    ),
  ),
);